### ISSUE TITLE
Upgrading to Next.js 13

### ISSUE THREAD

**Issue by spencerwooo** [spencerwooo/onedrive-vercel-index#780]
- [ ] Adapt new app directory and layout
- [ ] Rewrite of data rendering (also data fetching with React server components)
- [ ] Rewrite of Loading and Error layout
- [ ] ...

**Comment by aidenlx** [comment]
Using app directory and layout could help improving page loading significantly, since it can allow dynamicly load large libs like plyr, pdfjs, etc based on specific path. Currently we have to load everything since the website can't know the content for certain path ahead of time. 

Would you be interested in accepting PR regarding app directory refactoring, or collabrate on such refactoring in the future? I think the fetch API refactor on API route is the first step towards this goal, since the new data fetching system for server component is built on top of the native fetch() Web API.

**Comment by aidenlx** [comment]
one potential objective: [app/route](https://beta.nextjs.org/docs/routing/route-handlers) in favor of `pages/api` (available in nextjs 13.1.7, currently in canary)

**Comment by spencerwooo** [comment]
> Would you be interested in accepting PR regarding app directory refactoring, or collaborate on such refactoring in the future?

@aidenlx Certainly, but I don't think I can spare too much time on this project in the foreseeable future. If you are interested, go for it.
